testsuite/tests/typing-misc/empty_variant.ml

```diff
@@ -29,3 +29,37 @@ let f : t option -> int = function None -> 3
 [%%expect{|
 val f : t option -> int = <fun>
 |}]
+
+
+type nothing = |
+type ('a, 'b, 'c) t = | A of 'a | B of 'b | C of 'c
+module Runner : sig
```

Comment:
Concerning the test, a simpler
```ocaml
type nothing = |
type 'b t = A | B of 'b | C
let g (x:nothing t) = match x with A -> ()
```
triggers this issue. Did you have a specific reason in mind for the current test?

Reply:
This is a simplified version of the error that we observed in our code base -- that's how I derived it. No preference on whether we should use the simpler one or not.

Reply:
I think keep at least adding the simpler error case is useful, since it makes it clearer what is the intend of the test.